sys/kern/uio.c

```diff
@@ -81,6 +82,20 @@ int uiomove(void *buf, size_t n, uio_t *uio) {
   return error;
 }
 
+void uio_rollback(uio_t *uio, size_t nbytes) {
+  /* We assume that the `nbytes` supplied by the caller won't cause us to go
+   * past the beginning of the original uio->uio_iov.  */
```

Comment:
I think you should check this. If you leave it like it is someone will have trouble with finding this in the future. (Or maybe it  will be used only in `pty_write()` so we don't have to worry about it?)

Reply:
Checking this would require adding an additional field to `uio_t`. For instance, we could keep the number of bytes written so far.

Reply:
Yes. Why can't we add it?

Reply:
We can, I just wasn't sure if it's worth it. I'll add the extra field.